src/sentry/eventstream/kafka/backend.py

```diff
@@ -26,13 +30,73 @@ def delivery_callback(self, error, message):
         if error is not None:
             logger.warning("Could not publish message (error: %s): %r", error, message)
 
+    def _get_headers_for_insert(
+        self,
+        group,
+        event,
+        is_new,
+        is_regression,
+        is_new_group_environment,
+        primary_hash,
+        received_timestamp: float,
+        skip_consume,
+    ) -> Mapping[str, str]:
+
+        # HACK: We are putting all this extra information that is required by the
+        # post process forwarder into the headers so we can skip parsing entire json
+        # messages. The post process forwarder is currently bound to a single core.
+        # Once we are able to parallelize the JSON parsing and other transformation
+        # steps being done there we may want to remove this hack.
+        def encode_bool(value: Optional[bool]) -> str:
+            if value is None:
+                value = False
+            return str(int(value))
+
+        # WARNING: We must remove all None headers. There is a bug in confluent-kafka-python
+        # (used by both Sentry and Snuba) that incorrectly decrements the reference count of
+        # Python's None on any attempt to read header values containing null values, leading
+        # None to eventually get deallocated and crash the interpreter. The bug exists in the
+        # version we are using (1.5) as well as in the latest (at the time of writing) 1.7 version.
+        def strip_none_values(value: Mapping[str, Optional[str]]) -> Mapping[str, str]:
+            return {key: value for key, value in value.items() if value is not None}
+
+        send_new_headers = options.get("eventstream:kafka-headers")
+
+        if send_new_headers is True:
+            return strip_none_values(
+                {
+                    "Received-Timestamp": str(received_timestamp),
+                    "event_id": str(event.event_id),
+                    "project_id": str(event.project_id),
+                    "group_id": str(event.group_id) if event.group_id is not None else None,
+                    "primary_hash": str(primary_hash) if primary_hash is not None else None,
+                    "is_new": encode_bool(is_new),
+                    "is_new_group_environment": encode_bool(is_new_group_environment),
+                    "is_regression": encode_bool(is_regression),
```

Comment:
Do we ever make a distinction between False and None? This will conflate both cases and turn None into False
It seems to me these values can be None today and they would be propagated as None.
https://github.com/getsentry/sentry/blob/1584c198f01274017a87655ff98076cc6c00d14e/src/sentry/eventstream/kafka/protocol.py#L27-L28

Reply:
We don't, but I will change this to keep the original nullable values just in case

Reply:
Actually I'm going to leave this as is. I did a bit more testing - `eventstream.insert()` never gets passed None for any of these boolean values from the event_manager. The encode_bool function handles None anyway (but this is just being paranoid, it doesn't actually happen). Interestingly the `is_regression` value inside the message body is sometimes set to None and that was used, but the variable being passed never is. Since we don't read the message body all we can do is rely on the variable being passed in which is never None.